### DISCUSSION TITLE
Question about the coupling method used in Cardinal

### DISCUSSION THREAD

**Discussion by Ethan-xj** [neams-th-coe/cardinal#892] [Q&A]
I want to refer to the coupling method used in Cardinal to couple two external programs based on MOOSE, but the code of Cardinal is somewhat complex. Which part of the code should I look at to understand the specific coupling method? If possible, could you briefly introduce the coupling method between NekRS and OpenMC?

Ethan

**Comment by aprilnovak** [comment]
Hi @Ethan-xj, the code in Cardinal to send data in/out of NekRS and OpenMC is relatively simple, but can be hard to locate among the other code related to the solves. Can you tell me a bit more about the two external programs you have, so that I can best point you to the places in Cardinal which are relevant? NekRS has its data stored on a mesh, while OpenMC uses a geometry not based on a mesh, so I can save you some time in pointing you to either/both of these depending on how your external codes represent geometry and their solution fields.
